Test failure in `pre` job: test_wilcoxon_tie_correction started failing

### Please make sure these conditions are met

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of scanpy.
- [x] (optional) I have confirmed this bug exists on the main branch of scanpy.

### What happened?

Seems like some dependency update caused this to start failing, we need to investigate, maybe make upstream not release the offending change.

### Minimal code sample

```shell
hatch test -i deps=pre tests/test_rank_genes_groups.py::test_wilcoxon_tie_correction
```

### Error output

```pytb
FAILED tests/test_rank_genes_groups.py::test_wilcoxon_tie_correction[True] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 383 / 765 (50.1%)
Max absolute difference among violations: 2.4703574419e-07
Max relative difference among violations: 4.4790295425e-05
 ACTUAL: array([2.029091e-18, 6.430380e-01, 6.164863e-01, 7.551346e-02,
       1.590458e-08, 1.884037e-03, 5.290635e-01, 1.314833e-05,
       5.756106e-08, 2.646892e-01, 2.670054e-02, 4.151307e-01,...
 DESIRED: array([2.029090e-18, 6.430379e-01, 6.164863e-01, 7.551347e-02,
       1.590460e-08, 1.884038e-03, 5.290635e-01, 1.314831e-05,
       5.756110e-08, 2.646892e-01, 2.670054e-02, 4.151307e-01,...
FAILED tests/test_rank_genes_groups.py::test_wilcoxon_tie_correction[False] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 465 / 765 (60.8%)
Max absolute difference among violations: 7.0936388474e-07
Max relative difference among violations: 0.435111574
 ACTUAL: array([4.747381e-47, 2.910919e-01, 9.350926e-01, 1.204920e-01,
       6.826701e-18, 4.262317e-04, 6.489862e-01, 3.630759e-11,
       3.053055e-17, 3.037204e-01, 2.432239e-01, 5.370464e-01,...
 DESIRED: array([0.000000e+00, 2.910920e-01, 9.350926e-01, 1.204920e-01,
       6.826711e-18, 4.262317e-04, 6.489862e-01, 3.630755e-11,
       3.053048e-17, 3.037204e-01, 2.432239e-01, 5.370464e-01,...
```

### Versions

We should run session-info2 in CI to make this easier!

<details>

```
 + array-api-compat==1.12.0
 + contourpy==1.3.3
 + cycler==0.12.1
 + donfig==0.8.1.post1
 + fast-array-utils==1.3.1
 + fonttools==4.61.1
 + google-crc32c==1.8.0
 + h5py==3.15.1
 + joblib==1.5.3
 + kiwisolver==1.4.10rc0
 + legacy-api-wrap==1.5
 + llvmlite==0.46.0
 + matplotlib==3.10.8
 + natsort==8.4.0
 + networkx==3.6.1
 + numba==0.63.1
 + numcodecs==0.16.5
 + numpy==2.3.5
 + packaging==25.0
 + pandas==2.3.3
 + patsy==1.0.2
 + pillow==12.0.0
 + pynndescent==0.5.13
 + pyparsing==3.3.0b1
 + python-dateutil==2.9.0.post0
 + pytz==2025.2
 + pyyaml==6.0.3
 + scikit-learn==1.8.0
 + scipy==1.17.0rc1
 + seaborn==0.13.2
 + session-info2==0.2.3
 + six==1.17.0
 + statsmodels==0.14.6
 + threadpoolctl==3.6.0
 + tqdm==4.67.1
 + typing-extensions==4.15.0
 + tzdata==2025.3
 + umap-learn==0.5.9.post2
 + zarr==3.1.5
 + anndata==0.13.0.dev89+g0c7c2bfc2 (from git+https://github.com/scverse/anndata.git@0c7c2bfc2acc5a168ae3d4b0479f2a0b33d5e93f)
 + asttokens==3.0.1
 + click==8.3.1
 + cloudpickle==3.1.2
 + comm==0.2.3
 + coverage==7.13.0
 + coverage-enable-subprocess==1.0
 + dask==2025.12.0
 + dask-glm==0.3.2
 + dask-ml==2025.1.0
 + debugpy==1.8.19
 + decorator==5.2.1
 + dependency-groups==1.3.1
 + distributed==2025.12.0
 + execnet==2.1.2
 + executing==2.2.1
 + fsspec==2025.12.0
 + igraph==1.0.0
 + imageio==2.37.2
 + iniconfig==2.3.0
 + ipykernel==7.1.0
 + ipython==9.8.0
 + ipython-pygments-lexers==1.1.1
 + jedi==0.19.2
 + jinja2==3.1.6
 + jupyter-client==8.7.0
 + jupyter-core==5.9.1
 + lazy-loader==0.4
 + leidenalg==0.11.0
 + locket==1.0.0
 + markupsafe==3.0.3
 + matplotlib-inline==0.2.1
 + msgpack==1.1.2
 + multipledispatch==1.0.0
 + nest-asyncio==1.6.0
 + numpy-typing-compat==20251206.2.3
 + optype==0.15.0
 + parso==0.8.5
 + partd==1.4.2
 + pexpect==4.9.0
 + platformdirs==4.5.1
 + pluggy==1.6.0
 + prompt-toolkit==3.0.52
 + psutil==7.1.3
 + ptyprocess==0.7.0
 + pure-eval==0.2.3
 + pyarrow==22.0.0
 + pygments==2.19.2
 + pytest==9.0.2
 + pytest-cov==7.0.0
 + pytest-mock==3.15.1
 + pytest-randomly==3.16.0
 + pytest-rerunfailures==14.0
 + pytest-xdist==3.8.0
 + pyzmq==27.1.0
 ~ scanpy==1.13.0.dev32+g894bff5ff (from file:///home/runner/work/scanpy/scanpy)
 + scikit-image==0.26.0rc1
 + scikit-misc==0.5.2
 + scipy-stubs==1.16.3.3
 + sortedcontainers==2.4.0
 + sparse==0.17.0
 + stack-data==0.6.3
 + tblib==3.2.2
 + texttable==1.7.0
 + tifffile==2025.12.12
 + toolz==1.1.0
 + tornado==6.5.4
 + towncrier==25.8.0
 + traitlets==5.14.3
 + tuna==0.5.11
 + urllib3==2.6.2
 + wcwidth==0.2.14
 + zict==3.0.0
```

</details>

--- a/tests/test_rank_genes_groups.py
+++ b/tests/test_rank_genes_groups.py
@@ -267,8 +267,8 @@ def test_wilcoxon_symmetry():
     assert np.allclose(np.abs(stats_mono), np.abs(stats_dend))
 
 
-@pytest.mark.parametrize("reference", [True, False])
-def test_wilcoxon_tie_correction(reference):
+@pytest.mark.parametrize("reference", [True, False], ids=["ref", "rest"])
+def test_wilcoxon_tie_correction(*, reference: bool) -> None:
     pbmc = pbmc68k_reduced()
 
     groups = ["CD14+ Monocyte", "Dendritic"]
@@ -276,19 +276,20 @@ def test_wilcoxon_tie_correction(reference):
 
     _, groups_masks = select_groups(pbmc, groups, groupby)
 
-    x = pbmc.raw.X[groups_masks[0]].toarray()
+    if reference:
+        ref = groups[1]
+        mask_rest = groups_masks[1]
+    else:
+        ref = "rest"
+        mask_rest = ~groups_masks[0]
+        groups = groups[:1]
 
-    mask_rest = groups_masks[1] if reference else ~groups_masks[0]
+    assert isinstance(pbmc.raw.X, CSBase)
+    x = pbmc.raw.X[groups_masks[0]].toarray()
     y = pbmc.raw.X[mask_rest].toarray()
 
     pvals = mannwhitneyu(x, y, use_continuity=False, alternative="two-sided").pvalue
     pvals[np.isnan(pvals)] = 1.0
-
-    if reference:
-        ref = groups[1]
-    else:
-        ref = "rest"
-        groups = groups[:1]
 
     test_obj = _RankGenes(pbmc, groups, groupby, reference=ref)
     with (
@@ -298,7 +299,7 @@ def test_wilcoxon_tie_correction(reference):
     ):
         test_obj.compute_statistics("wilcoxon", tie_correct=True)
 
-    np.testing.assert_allclose(test_obj.stats[groups[0]]["pvals"], pvals)
+    np.testing.assert_allclose(test_obj.stats[groups[0]]["pvals"], pvals, atol=1e-6)
 
 
 def test_wilcoxon_huge_data(monkeypatch):

